src/guiapi/src/display_helper.cpp

```diff
@@ -31,103 +53,123 @@ void fill_till_end_of_line(const int left, const int top, const int h, Rect16 rc
-
-            if (y + h > rc.EndPoint().y) /// next char won't fit vertically
-                break;
-
-            continue;
+            break; /// end of single line => no more text to print
         }
 
         if (x + w > rc.EndPoint().x) {
             continue;
         }
 
         /// draw part
-#ifdef UNACCENT
-        // FIXME no check for enough space to draw char/chars
-        if (c < 128) {
-            display::DrawChar(point_ui16(x, y), c, pf, clr_bg, clr_fg);
-            x += w;
-        } else {
-            auto convertedChar = ConvertUnicharToFontCharIndex(c);
-            for (size_t i = 0; i < convertedChar.second; ++i) {
-                display::DrawChar(point_ui16(x, y), convertedChar.first[i], pf, clr_bg, clr_fg);
-                x += w; // this will screw up character counting for DE language @@TODO
-            }
-        }
-#else
-        display::DrawChar(point_ui16(x, y), c, pf, clr_bg, clr_fg);
-        x += w;
-#endif
+        draw_char_and_increment(pf, clr_bg, clr_fg, c, x, y, w);
     }
-    /// fill background to the end of the line and all below till the border of @rc
-    fill_till_end_of_line(x, y, h, rc, clr_bg);
-    y += h;
-    int h1 = std::max(0, rc.EndPoint().y - y);
-    if (h1 > 0) /// FIXME hotfix because FillRect draws nonempty rect. for height 0
-        display::FillRect(Rect16(rc.Left(), y, rc.Width(), h1), clr_bg);
 
     return size_ui16_t { rc.Width(), rc.Height() };
 }
 
-/// Fills space between two rectangles with a color
-/// @r_in must be completely in @r_out
-void fill_between_rectangles(const Rect16 *r_out, const Rect16 *r_in, color_t color) {
-    if (!r_out->Contain(*r_in))
-        return;
-    /// top
-    const Rect16 rc_t = { r_out->Left(), r_out->Top(), r_out->Width(), uint16_t(r_in->Top() - r_out->Top()) };
-    display::FillRect(rc_t, color);
-    /// bottom
-    const Rect16 rc_b = { r_out->Left(), int16_t(r_in->Top() + r_in->Height()), r_out->Width(), uint16_t((r_out->Top() + r_out->Height()) - (r_in->Top() + r_in->Height())) };
-    display::FillRect(rc_b, color);
-    /// left
-    const Rect16 rc_l = { r_out->Left(), r_in->Top(), uint16_t(r_in->Left() - r_out->Left()), r_in->Height() };
-    display::FillRect(rc_l, color);
-    /// right
-    const Rect16 rc_r = { int16_t(r_in->Left() + r_in->Width()), r_in->Top(), uint16_t((r_out->Left() + r_out->Width()) - (r_in->Left() + r_in->Width())), r_in->Height() };
-    display::FillRect(rc_r, color);
+/// Draws a text into the specified rectangle @rc
+/// If a character does not fit into the rectangle the drawing is stopped
+/// \param clr_bg background color
+/// \param clr_fg font/foreground color
+/// \returns size of drawn area
+/// Draws unused space of @rc with @clr_bg
+size_ui16_t render_text_singleline(Rect16 rc, string_view_utf8 str, const font_t *pf, color_t clr_bg, color_t clr_fg) {
+    no_wrap text_plain;
+
+    return render_line(text_plain, rc, str, pf, clr_bg, clr_fg);
+}
+
+struct dummy_font {
```

Comment:
please put this `dummy_font` to str_utils.hpp above the `RectTextLayout` declaration and use simple doxygen to describe a basic purpose. There is already similar structure
 `struct monospace {
    const uint32_t w = 12;
};`
so you can take it as inspiration